hack/lint-dependencies.sh

```diff
@@ -58,6 +58,19 @@ case "${1:-}" in
   ;;
 esac
 
+# List of dependencies we need to avoid dragging back into kubernetes/kubernetes
+forbidden_repos=(
+  "k8s.io/klog"  # we have switched to klog v2, so avoid klog v1
+)
+forbidden_repos=( "${forbidden_repos[@]/%/@}" ) # add the '@' suffix to we deal correct with modules with versions
+forbidden_repos_pattern=$(IFS="|"; echo "${forbidden_repos[*]}")
+deps_on_forbidden=$(go mod graph | grep -E "${forbidden_repos_pattern}" || echo "")
+if [ -n "${deps_on_forbidden}" ]; then
+  kube::log::error "The following have transistive dependencies on forbidden packages:"
+  echo "${deps_on_forbidden}"
+  exit 1
+fi
```

Comment:
it might be more helpful to iterate over the forbidden deps and list each set standalone:

```sh
rc=0
for forbidden_repo in "${forbidden_repos[@]}"; do
  deps_on_forbidden=$(go mod graph | grep " ${forbidden_repo}@" || echo "")
  if [ -n "${deps_on_forbidden}" ]; then
    kube::log::error "The following packages have transitive dependencies on ${forbidden_repo}, which is not allowed:"
    echo "${deps_on_forbidden}"
    echo ""
    rc=1
  fi
done

...

if [[ -n "${unused}${outdated}" ]]; then
  rc=1
fi

...

exit $rc
```

Reply:
Done!